framework/include/mesh/MooseMesh.h

```diff
@@ -171,6 +171,11 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
    */
   virtual unsigned int effectiveSpatialDimension() const;
 
+  /**
+   * Returns the maximum element dimension on the given blocks
+   */
+  virtual unsigned int getBlocksDimension(const std::vector<SubdomainName> & blocks) const;
```

Comment:
getBlocksMaxDimension

also why virtual?

Reply:
I was probably just (falsely) thinking that if the other dimension methods needed to be overridden, then this one may too, but I'll just keep it non-virtual until the need arises.